seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/SubPlan.java

```diff
@@ -231,6 +231,15 @@ private PipelineStatus getPipelineEndState() {
                 errorByPhysicalVertex.compareAndSet(
                         null, checkpointCoordinatorState.getThrowableMsg());
             }
+
+            // Because the pipeline state must update by tasks, If the pipeline can not get enough
+            // slot,
+            // the pipeline state will turn to Failing and then cancel all tasks in this pipeline.
+            // Because the tasks never run, so the tasks will complete with CANCELED. But the actual
+            // status of the pipeline should be FAILED
```

Comment:
```suggestion
            // Because the pipeline state must update by tasks, If the pipeline can not get enough
            // slot, the pipeline state will turn to Failing and then cancel all tasks in this pipeline.
            // Because the tasks never run, so the tasks will complete with CANCELED. But the actual
            // status of the pipeline should be FAILED
```